dask/array/tests/test_overlap.py

```diff
@@ -702,3 +723,55 @@ def test_trim_boundry(boundary):
     x_overlaped = da.overlap.overlap(x, 2, boundary=boundary)
     x_trimmed = da.overlap.trim_overlap(x_overlaped, 2, boundary=boundary)
     assert np.all(x == x_trimmed)
+
+
+def test_map_overlap_rechunks_array_if_needed():
+    # https://github.com/dask/dask/issues/6597
+    expected = np.arange(11)
+    x = da.from_array(expected, chunks=5)
+    y = x.map_overlap(lambda x: x, depth=2, boundary=0)
+    assert y.chunks == ((5, 6),)
+    assert_array_equal(y, expected)
+
+
+def test_map_overlap_rechunks_array_if_needed():
```

Comment:
Duplicate name with previous test.

Reply:
thanks! I think I didn't mean to have  one of those